sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py

```diff
@@ -78,3 +79,16 @@ def _is_cloud_event(event):
         return all([_ in event for _ in required]) and event['specversion'] == "1.0"
     except TypeError:
         return False
+
+def _deserialize_data(event, event_type):
+    """
+    Sets the data of the desrialized event to strongly typed event object if event type exists in _event_mappings.
+    Otherwise, sets it to None.
+
+    :param str event_type: The event_type of the EventGridEvent object or the type of the CloudEvent object.
+    """
+    # if system event type defined, set event.data to system event object
+    try:
+        event.data = (_event_mappings[event_type]).deserialize(event.data)
```

Comment:
Separately from the above, I've a bit of an inclination to think twice when  function isn't idempotent (worded less obtusely, if you run this twice on an event it'll fail and I'm not convinced that's necessary); I would imagine some sort of flow from a (potentially internal) "raw_data" field of some sort to generate progressively more structured types?